src/validation/transaction.cairo

```diff
@@ -135,12 +140,59 @@ pub fn validate_absolute_locktime(
     check_threshold_result
 }
 
+pub fn validate_relative_locktime(
```

Comment:
It makes sense to refactor this function to work with transaction input rather than the entire transaction. It would also be easier to test.

Reply:
i ask the question earlier, but i never got any response,

i will revert back to using transaction input

Reply:
`validate_absolute_locktime` requires access to the locktime field, hence it works with the entire transaction. But I agree that it might be better structured so that we pass only the data this function needs.  
We will probably refactor absolute/relative locktime validation afterwards to remove the overlap.